lib/HalModuleParser.js

```diff
@@ -245,7 +245,7 @@ HalModuleParser.prototype = {
 		var mightBeSystem = (sysModuleStartAddy.indexOf("80") === 0);
 		var isCore = ((userModuleStartAddy.indexOf("20") === 0) && (coreModuleStartAddy === "8005000"));
 		var mightBeBluz = (bluzModulesPlatformID == 103 && bluzModuleStartAddy < 262144);
-		var isMesh = [12, 13, 14].indexOf(meshModule.platformID)>=0 && (meshModule.moduleStartAddy === '30000' || meshModule.moduleStartAddy === 'f4000') &&
+		var isMesh = (meshModule.platformID >= 12 && meshModule.platformID <= 30) && (meshModule.moduleStartAddy === '30000' || meshModule.moduleStartAddy === 'f4000') &&
```

Comment:
So only issue with the reader was comparing the platform id to the new SOM ids?
I didn't see any changes in the actual reader logic

Reply:
The binary version reader has a bunch of heuristics to try to guess the location of the Particle firmware module header from a binary file. The logic should be flexible enough that it will work with all reasonable binaries but not too open so we don't get false positives identifying binaries of an older generation as newer or vice-versa.

The old logic was check for platforms 12, 13 and 14 explicitly. The new code checks for platforms between 12 and 30. Presumably future Gen 3 platforms would use 15, 16, etc and would use the same binary module format as the current platforms. So I think it's an OK compromise.